packages/jest-editor-support/src/Snapshot.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const fs = require('fs');
+const walker = require('./walker');
+const utils = require('jest-snapshot/build/utils');
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  name: string,
+  node: Node,
+  content?: string,
+};
+
+// create a lookup table from an array.
+const lookupFromArray = array => array.reduce(
```

Comment:
Is this really necessary?

```
const describeVariants = Object.assign(Object.create(null), {
  describe: true,
  fdescribe: true,
  xdescribe: true,
});
```

seems fine to me, without additional runtime overhead of looping over an array and modifying an object.